common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -664,4 +699,17 @@ private static void setCorrelationIdOnResult(@NonNull final CommandResult comman
         }
     }
 
+    private static void performRefresh(final SilentTokenCommand command) {
+        final SilentTokenCommandParameters params = ((SilentTokenCommandParameters) command.getParameters()).toBuilder().refreshDueToRefreshIn(true).correlationId(UUID.randomUUID().toString()).build();
+        Logger.info(
+                TAG,
+                "SilentTokenCommand with CorrelationId: "
+                        + command.getParameters().getCorrelationId()
+                        + " resulted in a REFRESH and initiated a RefreshCommand with CorrelationId: "
+                        + params.getCorrelationId()
+        );
+        final SilentTokenCommand refreshCommand = new SilentTokenCommand(params, command.getDefaultController(), command.getCallback(), command.getPublicApiId());
+        submitSilent(refreshCommand);
```

Comment:
Also it seems like we are going to be returning the result of this refresh back to the caller as well...do we want that? I thought we had discussed that the caller won't be notified of this i.e. we won't invoke their callback with results of refreshes...this could be confusing to the developers

Reply:
Good call.. Hmm.. We had discussed reusing the submitSilent call with Adam, @AdamBJohnsonx did you have an idea of how we could avoid returning to the client from the second silent call?

Reply:
We could have a CommandResult.REFRESH_COMPLETE that doesn't return back to the client. This way the initial submitSilent will return a REFRESH, and return the refresh-expired token to the client. Then the subsequent will return a REFRESH_COMPLETE, and the CommandDispatcher will know not to notify the client with a callback.

Reply:
I think what you want is to just use a different callback than the one that was provided by the client

Reply:
Yeah.. I like that better because it doesn't add changes to CommandResult